planet/src/main/java/com/google/planet/data/ItineraryGenerator.java

```diff
@@ -123,14 +117,56 @@ private int getTravelDurationInMinutes(Duration duration) {
                 items.add(item);
 
                 // Update the next event's start time
-                if (i != events.size() - 1) { 
-                    start += event.getDurationInMinutes() + getTravelDurationInMinutes(travelTimeGraph[i][i+1]);
-                }
+                start += event.getDurationInMinutes() + getTravelDurationInMinutes(directionsLegs[i].duration);
             }else {
                 errorMessage = "Sorry, you have too many events in a day!";
                 return Arrays.asList();
             }
         }
         return items;
     }
+
+    // Function that makes a request to the Directions API and return a direction route
+    // A DirectionsRoute has field legs, which is an array of DirectionsLegs
+    // Each DirectionsLeg contains starting address, ending address, and duration
+    // DirectionRoute also has int[] waypointOrder, which will be empty if optimized is set to false,
+    // and will contain the new order of waypoints if optimized is set to true
+    // For the MVP, real time traffic is NOT used.
+    private DirectionsRoute getDirectionsRoute(List<Event> events, boolean optimized) {
+        String GoogleApiKey = "AIzaSyDK36gDoYgOj4AlbCqh1IuaUuTlcpKF0ns";
```

Comment:
Make this final and consider storing this in a more permanent place (e.g. as a static member of a Constants class).
Should the API key even be here? There's an email titled "[STEP Internship] Reminder - Protect your API keys". It seems that we are not supposed to commit api keys into the repo. Instead, use .gitignore and environment variables to retrieve the key.

Reply:
This is the same key that needs to be used for the google maps javascript api, and since the google maps javascript api is on the client side, we can't really hide the key. The GCP console has an option to restrict the key, which only allows requests from specified websites (I think we already restricted the key? @emilyzhaoo )
I can go ahead and hide it in .gitignore if you think this is important, however it'll still show up in the html files for loading the map.

Reply:
I was also thinking about the same thing. Since we are asked to hide it, then I think we can try and **dynamically** add (using javascript) the element (script) that includes the key in the places.html file. I am not sure we can do that but we can try!

Reply:
@EkkiD Do you know more about this?

Reply:
As we discussed during today's stand up, please put the key in an ignored file so it doesn't get committed into github.

Reply:
Done!